### DISCUSSION TITLE
`pts_semantic_mask_path` key is not exist in the nuscenes dataset for occupancy prediction task

### DISCUSSION THREAD

**Discussion by SeaBird-Go** [open-mmlab/mmdetection3d#2587] [General] [answered]
Hello, when I run the `TPVFormer` project by the following command:
```bash
bash tools/dist_test.sh projects/TPVFormer/configs/tpvformer_8xb1-2x_nus-seg.py checkpoints/tpvformer_pretrained_fcos3d_r101_dcn.pth 8
```
I encountered the error that:
```bash
  File "~/mmdetection3d/mmdet3d/datasets/transforms/loading.py", line 1060, in transform
    results = self._load_semantic_seg_3d(results)
  File "~/mmdetection3d/mmdet3d/datasets/transforms/loading.py", line 945, in _load_semantic_seg_3d
    pts_semantic_mask_path = results['pts_semantic_mask_path']
KeyError: 'pts_semantic_mask_path'
```
It seems the `pts_semantic_mask_path` key not exist in the nuscenes dataset. And even though I create this dataset by following the **prepare dataset** step, it still fails.

**Comment by sunjiahao1999** [comment, marked as answer]
It seems to be that your `pkl` file does not have `pts_semantic_mask_path` in it. You can refer to this [guide](https://mmdetection3d.readthedocs.io/en/latest/advanced_guides/datasets/nuscenes.html) to regenerate `nuscenes_infos_train.pkl`. You have to make sure to additionally download the nuScenes lidarseg.

**Comment by sunjiahao1999** [comment]
btw, the checkpoint you are using is for training TPVFormer, you should use this checkpoint to validate the metrics.
![image](https://github.com/open-mmlab/mmdetection3d/assets/72679458/eb5aa16c-a326-43bb-a856-f3e302345980)
